internal/service/cognitoidp/user.go

```diff
@@ -0,0 +1,584 @@
+package cognitoidp
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUserCreate,
+		Read:   resourceUserRead,
+		Update: resourceUserUpdate,
+		Delete: resourceUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceUserImport,
+		},
+
+		// https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminCreateUser.html
+		Schema: map[string]*schema.Schema{
+			"attributes": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+			},
+			"client_metadata": {
+				Type:     schema.TypeMap,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"desired_delivery_mediums": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						cognitoidentityprovider.DeliveryMediumTypeSms,
+						cognitoidentityprovider.DeliveryMediumTypeEmail,
+					}, false),
+				},
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"force_alias_creation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"message_action": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					cognitoidentityprovider.MessageActionTypeResend,
+					cognitoidentityprovider.MessageActionTypeSuppress,
+				}, false),
+			},
+			"mfa_preference": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"sms_enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"software_token_enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"preferred_mfa": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Computed: true,
+			},
```

Comment:
We prefer matching the AWS API more closely. I would suggest replacing this with top-level parameters `mfa_setting_list` of type `schema.TypeSet` and `preferred_mfa_setting` of type `schema.TypeString`.

Since `MFAOptions` is no longer supported according to AWS API documentation, it can be ignored.